extensions/dacpac/src/test/dacpac.test.ts

```diff
@@ -45,3 +44,70 @@ describe('Sanitize database name for filename tests', function (): void {
+		should(isValidBasename(formatFileName('*.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('<.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('>.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('|.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('".dacpac'))).equal(isWindows ? false : true);
+
+		// Windows filenames cannot end with a whitespace
+		should(isValidBasename(formatFileName('test   .dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('test	.dacpac'))).equal(isWindows ? false : true);
+	});
+
+	it('Should determine Windows forbidden filenames', async () => {
+		// invalid only for Windows
+		should(isValidBasename(formatFileName('CON.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('PRN.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('AUX.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('NUL.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM1.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM2.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM3.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM4.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM5.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM6.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM7.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM8.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('COM9.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT1.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT2.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT3.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT4.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT5.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT6.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT7.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT8.dacpac'))).equal(isWindows ? false : true);
+		should(isValidBasename(formatFileName('LPT9.dacpac'))).equal(isWindows ? false : true);
+	});
+});
+
+function formatFileName(filename: string): string {
+	const filePath = isWindows ? 'c:\\users\\test\\' : '/Users/test/';
+	return filePath + filename;
```

Comment:
I'd probably just use os.tmpdir() to reduce the amount of hardcoded stuff like this we have